test/test-launcher/src/main/java/es/weso/shexsjava/Validate.java

```diff
@@ -0,0 +1,76 @@
+package es.weso.shexsjava;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.logging.Logger;
+
+import cats.effect.IO;
+import es.weso.asio.ontology.validate.ResultValidation;
+import es.weso.rdf.RDFReader;
+import es.weso.rdf.jena.RDFAsJenaModel;
+import es.weso.rdf.nodes.IRI;
+import es.weso.shapeMaps.ShapeMap;
+import es.weso.shex.ResolvedSchema;
+import es.weso.shex.Schema;
+import es.weso.shex.validator.Validator;
+import es.weso.utils.eitherios.EitherIOUtils;
+import scala.Option;
+
+/**
+ * Redefines shexsjava library (https://github.com/weso/shexsjava)
+ * @author Jose Emilio Labra Gayo
+ *
+ */
+public class Validate {
+
+	Logger log = Logger.getLogger(Validate.class.getName());
+
+	// none object is required to pass no base
+	Option<IRI> none = Option.empty();
+	Option<RDFReader> noneRDF = Option.empty();
+
+	public IO<ResultValidation> validate(String dataFile, String ontologyFile, String schemaFile, String shapeMapFile,
+			String expectedShapeMapFile) {
+		return readRDF(dataFile, "TURTLE").flatMap(rdfData -> readRDF(ontologyFile, "TURTLE")
+				.flatMap(ontologyData -> rdfData.merge(ontologyData).flatMap(merged -> getContents(schemaFile)
+						.flatMap(schemaStr -> Schema.fromString(schemaStr, "SHEXC", none, noneRDF).flatMap(
+								schema -> getShapeMap(shapeMapFile, merged, schema).flatMap(shapeMap -> ShapeMap
+										.fixShapeMap(shapeMap, merged, merged.getPrefixMap(), schema.prefixMap())
+										.flatMap(fixedShapeMap -> ResolvedSchema.resolve(schema, none).flatMap(
+												resolvedSchema -> getShapeMap(expectedShapeMapFile, merged, schema)
+														.flatMap(expectedShapeMap -> Validator
+																.validate(resolvedSchema, fixedShapeMap, merged)
+																.flatMap(result -> result.toResultShapeMap()
+																		.flatMap(resultShapeMap -> IO.pure(
+																				new ResultValidation(resultShapeMap,
+																						expectedShapeMap)))))))))))));
```

Comment:
I don't feel qualified to review _this_. Looks like too pure functional programming.